### PR TITLE
Fix #4420:Incorrect enableAttributeProfile value when exporting SAML service provider

### PR DESCRIPTION
**Pull request by IsurangaPerera**
Fix wso2/product-is#4420

### CODE REVIEW

File: components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/admin/SAMLSSOConfigAdmin.java [hunk 1]
```diff
@@ -355,6 +356,12 @@ public SAMLSSOServiceProviderInfoDTO getServiceProviders() throws IdentityExcept
                 providerDTO.setKeyEncryptionAlgorithmURI(providerDO.getKeyEncryptionAlgorithmUri());
                 providerDTO.setCertAlias(providerDO.getCertAlias());
                 providerDTO.setAttributeConsumingServiceIndex(providerDO.getAttributeConsumingServiceIndex());
+
+                if (providerDO.getAttributeConsumingServiceIndex() != null && !providerDO
+                        .getAttributeConsumingServiceIndex().equals("")) {
```
Comment (ashensw) [inline components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/admin/SAMLSSOConfigAdmin.java]: Better to use `StringUtils.isNotBlank()` here.

Comment (IsurangaPerera) [reply components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/admin/SAMLSSOConfigAdmin.java]: @ashensw Fixed

File: components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/admin/SAMLSSOConfigAdmin.java [hunk 2]
```diff
@@ -310,9 +310,9 @@ private SAMLSSOServiceProviderDTO createSAMLSSOServiceProviderDTO(SAMLSSOService
 
         serviceProviderDTO.setNameIDFormat(serviceProviderDO.getNameIDFormat());
 
-        if (serviceProviderDO.getAttributeConsumingServiceIndex() != null && !serviceProviderDO
-                .getAttributeConsumingServiceIndex().equals("")) {
+        if (StringUtils.isNotBlank(serviceProviderDO.getAttributeConsumingServiceIndex())) {
             serviceProviderDTO.setAttributeConsumingServiceIndex(serviceProviderDO.getAttributeConsumingServiceIndex());
+            serviceProviderDTO.setEnableAttributeProfile(true);
```
Comment (malithie) [inline components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/admin/SAMLSSOConfigAdmin.java]: I don't think this is the right fix to this issue.
Shouldn't the data object holding SAML related configs read from the persistence layer has this respective property being set.

Comment (IsurangaPerera) [reply components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/admin/SAMLSSOConfigAdmin.java]: At the time of DTO to DO mapping happens EnableAttributeProfile property is set to true depending on the AttributeConsumingServiceIndex value. So shouldn't we follow the same approach when mapping DO to DTO?

Comment (malithie) [reply components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/admin/SAMLSSOConfigAdmin.java]: +1 in that case. so it means this respective property is not persisted in the database and populated based on AttributeConsumingServiceIndex. Is that right

Comment (IsurangaPerera) [reply components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/admin/SAMLSSOConfigAdmin.java]: yeah, we have 2 option either modify DO or follow the same logic used in current implementation

Comment (malithie) [reply components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/admin/SAMLSSOConfigAdmin.java]: ok. agreed for the approach followed

**Comment by malithie** [comment]
@IsurangaPerera , can we implement unit test for this case

**Pull request merged**